loki/visitors/pretty_visualize.py

```diff
@@ -0,0 +1,349 @@
+        if comment:
+            return line + comment
+        return line
+
+    def visit_all(self, item, *args, **kwargs):
+        """
+        Convenience function to call :meth:`visit` for all given arguments.
+
+        If only a single argument is given that is iterable,
+        :meth:`visit` is called on all of its elements instead.
+        """
+        if is_iterable(item) and not args:
+            return list(
+                chain.from_iterable(
+                    as_tuple(self.visit(i, **kwargs) for i in item if i is not None)
+                )
+            )
+        return list(
+            chain.from_iterable(
+                as_tuple(
+                    self.visit(i, **kwargs) for i in [item, *args] if i is not None
+                )
+            )
+        )
+
+    def __add_node(self, node, **kwargs):
+        """
+        Adds a node to the graphical representation of the IR. Utilizes the
+        formatting provided by :meth:`format_node`.
+
+        Parameters
+        ----------
+        node: :any: `Node` object
+        kwargs["shape"]: str, optional (default: "oval")
+        kwargs["label"]: str, optional (default: format_node(repr(node)))
+        kwargs["parent"]: :any: `Node` object, optional (default: None)
+            If not available no edge is drawn.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            A list of a tuple of a node and potentially a edge information
+        """
+        label = kwargs.get("label", "")
+        if label == "":
+            label = self.format_node(repr(node))
+
+        shape = kwargs.get("shape", "oval")
+        label = (
+            label + "‎"
+        )  # dirty hack to force graphviz to utilize parenthesis around the label
+
+        node_key = str(id(node))
+        if node_key not in self._id_map:
+            self._id_map[node_key] = str(self._id)
+            self._id += 1
+
+        node_info = {
+            "name": str(self._id_map[node_key]),
+            "label": str(label),
+            "shape": str(shape),
+        }
+
+        parent = kwargs.get("parent")
+        edge_info = {}
+        if parent:
+            parent_id = self._id_map[str(id(parent))]
+            child_id = self._id_map[str(id(node))]
+            edge_info = {"tail_name": str(parent_id), "head_name": str(child_id)}
+
+        return [(node_info, edge_info)]
+
+    # Handler for outer objects
+    def visit_Module(self, o, **kwargs):
+        """
+        Add a :any:`Module`, mark parent node and visit all "spec" and "subroutine" nodes.
+
+        Returns
+        -------
+        list[tuple[dict[str,str], dict[str,str]]]]
+            An extended list of tuples of a node and potentially a edge information
+        """
+        node_edge_info = self.__add_node(o, **kwargs)
+        kwargs["parent"] = o
+
+        node_edge_info.extend(self.visit(o.spec, **kwargs))
+        node_edge_info.extend(self.visit(o.subroutines, **kwargs))
```

Comment:
```suggestion
        node_edge_info.extend(self.visit(o.contains, **kwargs))
```
The `contains` part can contain also pragmas, comments etc. which are omitted if picking out the subroutines only

Reply:
Resolved by 98d8d7842942cc505ac8f041344b852234d1952a